File: src/main/java/com/gtladd/gtladditions/api/machine/data/MachineEnergyData.java
package com.gtladd.gtladditions.api.machine.data;

import com.gregtechceu.gtceu.api.machine.MetaMachine;

import java.math.BigInteger;
import java.util.UUID;

public record MachineEnergyData(UUID userId, MetaMachine machine, BigInteger euPerTick)
        implements Comparable<MachineEnergyData> {

    @Override
    public int compareTo(MachineEnergyData other) {
        return this.euPerTick.compareTo(other.euPerTick);
    }
}

File: src/main/java/com/gtladd/gtladditions/common/data/MachineEnergyData.kt
package com.gtladd.gtladditions.common.data

import com.gregtechceu.gtceu.api.machine.MetaMachine
import java.math.BigInteger
import java.util.*

@JvmRecord
data class MachineEnergyData(@JvmField val userId: UUID, @JvmField val machine: MetaMachine, @JvmField val euPerTick: BigInteger) :
    Comparable<MachineEnergyData> {
    override fun compareTo(other: MachineEnergyData): Int {
        return this.euPerTick.compareTo(other.euPerTick)
    }
}
